addons/claim_from_delivery/models/stock_picking.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    claim_count_out = fields.Integer(compute='_compute_claim_count_out', string='Claims')
+
+    @api.multi
+    def _compute_claim_count_out(self):
+        refs = [('stock.picking,' + str(picking_id)) for picking_id in self.ids]
+        claim_data = self.env['crm.claim'].read_group([('ref', 'in', refs)], ['ref'], ['ref'])
+        claim_dict = {int(data['ref'].split(',')[1]): data['ref_count'] for data in claim_data}
+        for data in self:
+            data.claim_count_out = claim_dict.get(data.id)
```

Comment:
In this method 3 times ***loop*** and 1 ***read_group*** use try to convert in optimize it.